### PR TITLE
docs: update `hmac-auth` plugin doc per the refactoring

### PR DESCRIPTION
**Pull request by kayx23**
### Description

feat PR: https://github.com/apache/apisix/pull/11581

the doc is to be mentioned in release notes

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by moonming — DISMISSED** [review]

File: docs/en/latest/plugins/hmac-auth.md [hunk 1]
```diff
@@ -63,312 +69,474 @@ admin_key=$(yq '.deployment.admin.admin_key[0].key' conf/config.yaml | sed 's/"/
...
-:::
+You should see the request headers printed:
 
-**The algorithm for generating `canonical_query_string` is described below:**
-
-1. Extract the query terms from the URL.
-2. Split the query terms into key-value pairs by using `&` as the separator.
-3. If `encode_uri_params` is `true`:
-   1. If there are only keys, the conversion formula is `uri_encode(key) + "="`.
-   2. If there are both keys and values, the conversion formula is `uri_encode(key) + "=" + uri_encode(value)`. Here, the value can even be an empty string.
```
Comment (pottekkat) [inline docs/en/latest/plugins/hmac-auth.md:133]: Is this algorithm documented anywhere else or is it no longer needed?

File: docs/en/latest/plugins/hmac-auth.md [hunk 2]
```diff
@@ -30,31 +30,37 @@ description: This document contains information about the Apache APISIX hmac-aut
...
+| signed_headers        | array[string] | False    |               |                                             | The list of HMAC-signed headers that should be included in the client request's HMAC signature.  |
+| validate_request_body | boolean       | False    | false         |                              | If true, validate the integrity of the request body to ensure it has not been tampered with during transmission. Specifically, the plugin creates a SHA-256 base64-encoded digest and compare it to the `Digest` header. If the Digest` header is missing or if the digests do not match, the validation fails.                          |
+| hide_credentials | boolean       | False    | false         |                              | If true, do not pass the authorization request header to upstream services.                        |
 
-## Enable Plugin
+NOTE: `encrypt_fields = {"secret_key"}` is also defined in the schema, which means that the field will be stored encrypted in etcd. See [encrypted storage fields](../plugin-develop.md#encrypted-storage-fields).
```
[the remark this answers is not in the document]
Comment (kayx23) [reply docs/en/latest/plugins/hmac-auth.md:58]: I didn't change this from the original wording. This note is in most plugin docs. If this needs any change it's better to go in a different PR.

>  just say that the secret key will be encrypted before stored in etcd

this is what goes in the api7 docs yes

[the remark this answers is not in the document]
Comment (pottekkat) [reply docs/en/latest/plugins/hmac-auth.md:58]: Sounds good.

File: docs/en/latest/plugins/hmac-auth.md (hunk 1 above)
Comment (kayx23) [reply docs/en/latest/plugins/hmac-auth.md:133]: @Revolyssup hello could you help evaluate if the above about constructing `canonical_query_string` is still relevant? I can't find anything about "query string" in the RFC.

**Review by pottekkat — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**